### ISSUE TITLE
2.6.13 gets stuck in the Select Game screen

### ISSUE THREAD

**Issue by RogerCooper** [triplea-game/triplea#8758]
### How can the problem be recreated?

Run TripleA. Click "Select Game". After a long wait, I received a java.lang.NullPointerException.
### Do you have any ideas for an expected fix?
The recent changes had already slowed the select game. 
### If playing a prerelease, which version are you using?

Game Version:

### If playing a prerelease, does this happen on the latest release?

- [] yes
- [] no

**Comment by DanVanAtta** [comment]
Do you get an option to report to TripleA? Have you clicked that to submit the stack trace? Could you attach the file `<your_home_folder>/triplea/triplea.log` to this ticket?

**Comment by DanVanAtta** [comment]
Can your provide a way to recreate the downloaded maps that you have @RogerCooper so that this can be reproduced?

**Comment by RogerCooper** [comment]
Log files still don't generate, because you are trying to write them into a protected area. You should put them into the downloadedmaps folder. Here is the stack trace.
2.6.13@57528d46ee: MapDescriptionYamlGenerator#lambda$readMapNameFromXmlTags$0:143 - java.lang.NullPointerException

## User Description
See issue 8758

## TripleA Version
2.6.13@57528d46ee

## Java Version
11.0.9.1

## Operating System
Windows 10

## Memory
Heap utilization statistics [MB]
Used Memory: 22
Free memory: 1897
Total memory: 1919
Max memory: 1919


## Stack Trace
```
Exception: java.lang.NullPointerException
java.lang.Exception
	at org.triplea.map.description.file.MapDescriptionYamlGenerator.lambda$readMapNameFromXmlTags$0(MapDescriptionYamlGenerator.java:143)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:176)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1632)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:548)
	at org.triplea.map.description.file.MapDescriptionYamlGenerator.readMapNameFromXmlTags(MapDescriptionYamlGenerator.java:145)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1632)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findAny(ReferencePipeline.java:548)
	at org.triplea.map.description.file.MapDescriptionYamlGenerator.findAnyXmlAndReadMapName(MapDescriptionYamlGenerator.java:124)
	at org.triplea.map.description.file.MapDescriptionYamlGenerator.generateYamlDataForMap(MapDescriptionYamlGenerator.java:57)
	at org.triplea.map.description.file.MapDescriptionYaml.generateForMap(MapDescriptionYaml.java:89)
	at games.strategy.engine.framework.map.file.system.loader.DownloadedMaps.lambda$readMapYamlsAndGenerateMissingMapYamls$0(DownloadedMaps.java:44)
	at java.base/java.util.Optional.or(Optional.java:318)
	at games.strategy.engine.framework.map.file.system.loader.DownloadedMaps.lambda$readMapYamlsAndGenerateMissingMapYamls$1(DownloadedMaps.java:44)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:195)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:177)
	at java.base/java.util.AbstractList$RandomAccessSpliterator.forEachRemaining(AbstractList.java:720)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:484)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:913)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at games.strategy.engine.framework.map.file.system.loader.DownloadedMaps.readMapYamlsAndGenerateMissingMapYamls(DownloadedMaps.java:47)
	at games.strategy.engine.framework.map.file.system.loader.DownloadedMaps.<init>(DownloadedMaps.java:26)
	at games.strategy.engine.framework.map.file.system.loader.DownloadedMaps.parseMapFiles(DownloadedMaps.java:34)
	at games.strategy.engine.framework.ui.background.BackgroundTaskRunner$1.doInBackground(BackgroundTaskRunner.java:115)
	at java.desktop/javax.swing.SwingWorker$1.call(SwingWorker.java:304)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.desktop/javax.swing.SwingWorker.run(SwingWorker.java:343)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)


```


I will see if I can get you my full folder contents, but it will be a large file.

**Comment by RogerCooper** [comment]
Here is a link to the full folder contents https://drive.google.com/file/d/1HSq9E7du1MB4IrvBwZBQF90xl8zpy2_Y/view?usp=sharing

**Comment by DanVanAtta** [comment]
Thanks for the stack trace, that's going to help. I'll try the zip as well if that repro's. Last night I pulled all of the zips from rogercooper.com and found a few other problems but not the NPE.

`<your_home_folder>/triplea/` is protected on your system @RogerCooper ? AFAIK the game creates that folder to begin with. We were publishing logs to the installation folder which was protected, to avoid that we updated it to publish logs into that location. Please confirm you're not seeing the log file there, that would be a problem (if we can't write to '/triplea/', which is created by TripleA, there would be no guarantee we can create the downloadedMaps folder or write to any of the subfolders, seemingly quite the problem).

**Comment by DanVanAtta** [comment]
With the stack trace it was straight forward to see where the problem was. The following PR should fix this issue: https://github.com/triplea-game/triplea/pull/8781

**Comment by RogerCooper** [comment]
By default, TripleA installs to c:\Program Files\TripleA. This is a protected area in Windows and I must give permission when installing. This is normal for Windows programs. That is bad place to write log files. DownloadedMaps does not require special permission to write to.

rogercooper.com contains a fair amount of old unrunnable files and non-TripleA stuff. Which probably makes even better for testing.

**Comment by DanVanAtta** [comment]
Funny enough, most of the maps that I tried from rogercooper.com did work (and a number do look interesting!). There were a few issues, small percentage, and mostly missing images.

Agree on `c:\Program Files\` @RogerCooper , though please note that the path I mentioned was: `<your_home_folder>/triplea/`, that is not `c:\program Files\`. The triplea.log file should appear parallel to the downloadedMaps folder, both are in `<your_home_folder>/triplea/`

**Comment by RogerCooper** [comment]
I see the log file now. It is 1.36GB.

Now that I am a map admin, I will putting posting most of the files after they are reformatted for the current version of TripleA.

**Comment by DanVanAtta** [comment]
Hmm, I would have thought it should have been rotated before getting that large. I'll need to investigate that config.

It is expected to be particularly large, there is some really verbose logging in place to debug map parsing problems @Navalland2  has been observing. I'm looking forward to dialing that verbosity down.

**Comment by RogerCooper** [comment]
When do you expect to deploy the fix? The latest prerelease is still 2.6.13.

**Comment by DanVanAtta** [comment]
Thanks for asking, I was surprised there was not one. There were a few problems that are fixed now.
